File: src/screens/StoreScreen.js
import { useMemo } from 'react';
import { View, Text, StyleSheet, ScrollView } from 'react-native';
import { MaterialCommunityIcons } from '@expo/vector-icons';
import * as Haptics from 'expo-haptics';
import FadeInView from '../components/FadeInView';
import PressableScale from '../components/PressableScale';
import { useTheme } from '../ThemeContext';
import { themeList, spacing, radius, glowShadow } from '../theme';

// Prezzi mock: lo store è pronto per integrare pagamenti reali (Play Billing / RevenueCat).
const PRICES = {
  neonGreen: '2,99 €',
  neonCyan: '2,99 €',
  neonPink: '2,99 €',
  neonPurple: '2,99 €',
};

export default function StoreScreen() {
  const { theme, changeTheme, unlock, isUnlocked, themeId } = useTheme();
  const styles = useMemo(() => makeStyles(theme.colors), [theme]);

  const premium = themeList.filter((t) => t.premium);

  return (
    <ScrollView style={styles.container} contentContainerStyle={styles.content}>
      <View style={styles.banner}>
        <MaterialCommunityIcons name="shopping" size={28} color={theme.colors.primary} />
        <View style={{ flex: 1 }}>
          <Text style={styles.bannerTitle}>Neon Store</Text>
          <Text style={styles.bannerText}>
            Sblocca skin neon esclusive con effetti glow.
          </Text>
        </View>
      </View>

      <Text style={styles.section}>Skin Neon</Text>
      {premium.map((t, idx) => {
        const owned = isUnlocked(t.id);
        const active = themeId === t.id;
        return (
          <FadeInView
            key={t.id}
            delay={idx * 80}
            style={[
              styles.card,
              { borderColor: t.glow },
              t.neon && glowShadow(t.glow, 10),
            ]}
          >
            <View style={styles.preview}>
              <View style={[styles.swatch, { backgroundColor: t.colors.bg }]} />
              <View style={[styles.swatch, { backgroundColor: t.colors.card }]} />
              <View
                style={[
                  styles.swatch,
                  { backgroundColor: t.colors.primary },
                  glowShadow(t.glow, 8),
                ]}
              />
            </View>
            <View style={{ flex: 1 }}>
              <Text style={[styles.name, { color: t.glow }]}>{t.name}</Text>
              <Text style={styles.desc}>{t.description}</Text>
            </View>
            {owned ? (
              <PressableScale
                style={[styles.btn, active ? styles.btnActive : styles.btnOwned]}
                onPress={() => !active && changeTheme(t.id)}
                disabled={active}
              >
                <Text style={styles.btnText}>{active ? 'Attivo' : 'Applica'}</Text>
              </PressableScale>
            ) : (
              <PressableScale
                style={[styles.btn, { backgroundColor: t.glow }]}
                onPress={() => {
                  Haptics.notificationAsync(Haptics.NotificationFeedbackType.Success);
                  unlock(t.id);
                  changeTheme(t.id);
                }}
              >
                <Text style={styles.btnText}>{PRICES[t.id] || 'Sblocca'}</Text>
              </PressableScale>
            )}
          </FadeInView>
        );
      })}

      <Text style={styles.note}>
        Nota: gli acquisti sono dimostrativi (sblocco immediato). L'infrastruttura è
        pronta per collegare i pagamenti reali di App Store e Google Play.
      </Text>
    </ScrollView>
  );
}

const makeStyles = (c) =>
  StyleSheet.create({
    container: { flex: 1, backgroundColor: c.bg },
    content: { padding: spacing.md },
    banner: {
      flexDirection: 'row',
      alignItems: 'center',
      gap: spacing.md,
      backgroundColor: c.card,
      borderRadius: radius.md,
      padding: spacing.md,
      borderWidth: 1,
      borderColor: c.border,
    },
    bannerTitle: { color: c.text, fontSize: 18, fontWeight: '800' },
    bannerText: { color: c.textMuted, fontSize: 13, marginTop: 2 },
    section: {
      color: c.text,
      fontSize: 16,
      fontWeight: '700',
      marginTop: spacing.lg,
      marginBottom: spacing.sm,
    },
    card: {
      flexDirection: 'row',
      alignItems: 'center',
      gap: spacing.md,
      backgroundColor: c.card,
      borderRadius: radius.md,
      padding: spacing.md,
      marginBottom: spacing.md,
      borderWidth: 1.5,
    },
    preview: { flexDirection: 'row', gap: 3 },
    swatch: {
      width: 16,
      height: 40,
      borderRadius: 4,
    },
    name: { fontSize: 16, fontWeight: '800' },
    desc: { color: c.textMuted, fontSize: 12, marginTop: 2 },
    btn: {
      paddingHorizontal: spacing.md,
      paddingVertical: spacing.sm,
      borderRadius: radius.md,
      minWidth: 84,
      alignItems: 'center',
    },
    btnOwned: { backgroundColor: c.cardAlt },
    btnActive: { backgroundColor: c.border },
    btnText: { color: '#fff', fontWeight: '800' },
    note: { color: c.textMuted, fontSize: 12, lineHeight: 18, marginTop: spacing.sm },
  });

File: lib/screens/store_screen.dart
import 'package:flutter/material.dart';
import 'package:provider/provider.dart';
import '../state/theme_provider.dart';
import '../state/locale_provider.dart';
import '../services/feedback_service.dart';
import '../services/app_icon_service.dart';
import '../services/monetization_service.dart';
import '../services/analytics_service.dart';
import '../theme/app_theme.dart';
import '../widgets/emblem.dart';

class StoreScreen extends StatelessWidget {
  const StoreScreen({super.key});

  // Temi considerati "leggendari" (gli altri premium sono palette Neon).
  static const _legendaryIds = {
    'spartacus', 'kratos', 'ulisse', 'zeus', 'cyberpunk',
    'valkyrie', 'ronin', 'anubis', 'achille', 'leonida', 'poseidon',
    'ercole', 'odino', 'ra', 'ade',
    'cavaliere', 'cerberus', 'igris', 'sukuna', 'toji',
  };

  @override
  Widget build(BuildContext context) {
    final theme = context.watch<ThemeProvider>();
    final c = theme.colors;
    final t = context.watch<LocaleProvider>().t;

    final premium = themeList.where((s) => s.premium).toList();
    final legendary = premium.where((s) => _legendaryIds.contains(s.id)).toList()
      ..sort((a, b) => a.name.compareTo(b.name));
    final neon = premium.where((s) => !_legendaryIds.contains(s.id)).toList()
      ..sort((a, b) => a.name.compareTo(b.name));

    final locked = premium.where((s) => !theme.isUnlocked(s.id)).toList();

    return Scaffold(
      appBar: AppBar(title: Text(t('neon_store'))),
      body: ListView(
        padding: const EdgeInsets.all(Spacing.md),
        children: [
          _banner(context, c, t.call, locked),
          if (legendary.isNotEmpty) ...[
            _sectionHeader(c, t('theme_tab_legendary'), legendary.length),
            ...legendary.map((s) => _tile(context, theme, c, t.call, s)),
          ],
          if (neon.isNotEmpty) ...[
            const SizedBox(height: Spacing.sm),
            _sectionHeader(c, t('theme_tab_neon'), neon.length),
            ...neon.map((s) => _tile(context, theme, c, t.call, s)),
          ],
          const SizedBox(height: Spacing.md),
          Text(t('demo_purchases'),
              style: TextStyle(color: c.textMuted, fontSize: 12)),
        ],
      ),
    );
  }

  Widget _banner(BuildContext context, AppColors c, String Function(String) t,
      List<AppSkin> locked) {
    return Container(
      margin: const EdgeInsets.only(bottom: Spacing.md),
      padding: const EdgeInsets.all(Spacing.md),
      decoration: BoxDecoration(
        color: c.card,
        borderRadius: BorderRadius.circular(Radii.lg),
        border: Border.all(color: c.border),
      ),
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.start,
        children: [
          Text(t('store_banner'),
              style: TextStyle(color: c.text, fontWeight: FontWeight.w600)),
          if (locked.isNotEmpty) ...[
            const SizedBox(height: Spacing.sm),
            SizedBox(
              width: double.infinity,
              child: FilledButton.icon(
                style: FilledButton.styleFrom(backgroundColor: c.primary),
                onPressed: () async {
                  final themeProvider = context.read<ThemeProvider>();
                  FeedbackService.onUnlock();
                  // Se RevenueCat è configurato (API key presente) avvia un
                  // acquisto reale; altrimenti mantiene lo sblocco locale (demo).
                  if (MonetizationService.isAvailable) {
                    final ok = await MonetizationService.buyPremium();
                    if (!ok) return; // acquisto annullato o fallito
                    AnalyticsService.storeUnlock('all', paid: true, price: 9.99);
                  } else {
                    AnalyticsService.storeUnlock('all', paid: false);
                  }
                  themeProvider.unlockMany(locked.map((s) => s.id));
                },
                icon: const Icon(Icons.lock_open_rounded, size: 18),
                label: Text('${t('store_unlock_all')}  ·  9,99 €'),
              ),
            ),
          ],
        ],
      ),
    );
  }

  Widget _sectionHeader(AppColors c, String label, int count) {
    return Padding(
      padding: const EdgeInsets.only(top: Spacing.sm, bottom: Spacing.sm),
      child: Row(
        children: [
          Text(label.toUpperCase(),
              style: TextStyle(
                  color: c.textMuted,
                  fontSize: 12,
                  fontWeight: FontWeight.w800,
                  letterSpacing: 1.2)),
          const SizedBox(width: 6),
          Text('$count',
              style: TextStyle(color: c.textMuted, fontSize: 12)),
        ],
      ),
    );
  }

  Widget _tile(BuildContext context, ThemeProvider theme, AppColors c,
      String Function(String) t, AppSkin skin) {
    final owned = theme.isUnlocked(skin.id);
    final active = theme.themeId == skin.id;
    final glow = skin.glow ?? skin.colors.primary;
    final subject = emblemForTheme(skin.id);
    final hasIcon = AppIconService.styles
        .any((ic) => ic.premium && ic.themeId == skin.id);

    void buy() {
      FeedbackService.onUnlock();
      theme.unlock(skin.id);
      theme.changeTheme(skin.id);
    }

    void apply() {
      FeedbackService.onTap();
      theme.changeTheme(skin.id);
    }

    return Container(
      margin: const EdgeInsets.only(bottom: Spacing.sm),
      padding: const EdgeInsets.all(Spacing.sm),
      decoration: BoxDecoration(
        color: skin.colors.card,
        borderRadius: BorderRadius.circular(Radii.md),
        border: Border.all(
            color: active ? glow : glow.withValues(alpha: 0.35),
            width: active ? 2 : 1),
      ),
      child: Row(
        children: [
          // Anteprima
          Container(
            width: 46,
            height: 46,
            decoration: BoxDecoration(
              gradient: LinearGradient(
                begin: Alignment.topLeft,
                end: Alignment.bottomRight,
                colors: [skin.colors.cardAlt, skin.colors.bg],
              ),
              borderRadius: BorderRadius.circular(Radii.sm),
              border: Border.all(color: glow.withValues(alpha: 0.5)),
            ),
            clipBehavior: Clip.antiAlias,
            child: subject != null
                ? Center(
                    child: EmblemView(
                        subject: subject, size: 30, color: glow))
                : Center(
                    child: Icon(Icons.palette_rounded,
                        color: glow, size: 22)),
          ),
          const SizedBox(width: Spacing.sm),
          // Testo
          Expanded(
            child: Column(
              crossAxisAlignment: CrossAxisAlignment.start,
              children: [
                Text(skin.name,
                    maxLines: 1,
                    overflow: TextOverflow.ellipsis,
                    style: TextStyle(
                        color: skin.colors.text,
                        fontSize: 15,
                        fontWeight: FontWeight.w800)),
                Text(
                  hasIcon
                      ? '${skin.description} · ${t('store_icon_included')}'
                      : skin.description,
                  maxLines: 2,
                  overflow: TextOverflow.ellipsis,
                  style: TextStyle(color: skin.colors.textMuted, fontSize: 11),
                ),
              ],
            ),
          ),
          const SizedBox(width: Spacing.sm),
          // Azione
          _actionButton(t, glow, owned, active, buy, apply),
        ],
      ),
    );
  }

  Widget _actionButton(String Function(String) t, Color glow, bool owned,
      bool active, VoidCallback buy, VoidCallback apply) {
    if (active) {
      return Container(
        padding: const EdgeInsets.symmetric(horizontal: 12, vertical: 8),
        child: Row(mainAxisSize: MainAxisSize.min, children: [
          Icon(Icons.check_circle_rounded, size: 16, color: glow),
          const SizedBox(width: 4),
          Text(t('active'),
              style: TextStyle(color: glow, fontWeight: FontWeight.w700)),
        ]),
      );
    }
    if (owned) {
      return OutlinedButton(
        style: OutlinedButton.styleFrom(
            foregroundColor: glow,
            side: BorderSide(color: glow),
            visualDensity: VisualDensity.compact),
        onPressed: apply,
        child: Text(t('apply')),
      );
    }
    return FilledButton(
      style: FilledButton.styleFrom(
          backgroundColor: glow,
          visualDensity: VisualDensity.compact,
          padding: const EdgeInsets.symmetric(horizontal: 14)),
      onPressed: buy,
      child: const Text('2,99 €'),
    );
  }
}
